src/Httpstatus.php

```diff
@@ -157,15 +184,152 @@ public function __construct($statusArray = [])
     public function count()
     {
         return count($this->httpStatus);
     }
 
+    /**
+     * {@inheritdoc}
+     */
     public function getIterator()
     {
         return new ArrayIterator($this->httpStatus);
     }
+
+    /**
+     * Tell whether the status code is informational
+     *
+     * @param int $code
+     *
+     * @return bool
+     *
+     * @see https://tools.ietf.org/html/rfc7231#section-6.2
+     */
+    public function isInformational($code)
+    {
+        return (bool) $this->formatInt($this->filterHttpStatusCode($code), 100, 199);
+    }
+
+    /**
+     * Validate an integer between ranges
+     *
+     * @param int $int the integer to valid and format
+     * @param int $min the min range
+     * @param int $max the max range
+     *
+     * @return int|false return false if the integer is not valid
+     */
+    protected function formatInt($int, $min, $max)
+    {
+        return filter_var($int, FILTER_VALIDATE_INT, ['options' => [
+            'min_range' => $min,
+            'max_range' => $max,
+        ]]);
+    }
+
+    /**
+     * Tell whether the submitted code is successful
+     *
+     * @param int $code
+     *
+     * @return bool
+     *
+     * @see https://tools.ietf.org/html/rfc7231#section-6.3
+     */
+    public function isSuccessful($code)
+    {
+        return (bool) $this->formatInt($this->filterHttpStatusCode($code), 200, 299);
+    }
+
+    /**
+     * Tell whether the submitted code is for redirection
+     *
+     * @param int $code
+     *
+     * @return bool
+     *
+     * @see https://tools.ietf.org/html/rfc7231#section-6.4
+     */
+    public function isRedirection($code)
+    {
+        return (bool) $this->formatInt($this->filterHttpStatusCode($code), 300, 399);
+    }
+
+    /**
+     * Tell whether the submitted code is a client error
+     *
+     * @param int $code
+     *
+     * @return bool
+     *
+     * @see https://tools.ietf.org/html/rfc7231#section-6.5
+     */
+    public function isClientError($code)
+    {
+        return (bool) $this->formatInt($this->filterHttpStatusCode($code), 400, 499);
+    }
+
+    /**
+     * Tell whether the submitted code is a server error
+     *
+     * @param int $code
+     *
+     * @return bool
+     *
+     * @see https://tools.ietf.org/html/rfc7231#section-6.6
+     */
+    public function isServerError($code)
+    {
+        return (bool) $this->formatInt($this->filterHttpStatusCode($code), 500, 599);
+    }
+
+    /**
+     * Tell whether the submitted code is a custom status code
+     *
+     * @param int $code
+     *
+     * @return bool
+     */
+    public function isCustom($code)
+    {
+        return 600 <= $this->filterHttpStatusCode($code);
+    }
+
+    /**
+     * Tell whether the submitted code is a unused status code By IANA
+     *
+     * @param int $code
+     *
+     * @return bool
+     */
+    public function isUnused($code)
+    {
+        return 306 === $this->filterHttpStatusCode($code);
```

Comment:
I'm not sure I like having a single method checking for a specific status code.  We don't have methods like `isOK()` or `isTemporaryRedirect()`, so I'd vote against something like this.